.github/workflows/test.yml

```diff
@@ -70,17 +82,21 @@ jobs:
           key: v1-${{ runner.os }}-gradle-${{ hashfiles('**/gradle-wrapper.properties') }}-${{ hashFiles('**/*.gradle') }}
           restore-keys: v1-${{ runner.os }}-gradle-${{ hashfiles('**/gradle-wrapper.properties') }}
       - name: Grant execute permission for gradlew
+        if: steps.skiptest.outputs.is-bump == 'no'
         run: chmod +x gradlew
       - name: Check Javadoc
+        if: steps.skiptest.outputs.is-bump == 'no'
         run: ./gradlew javadoc --scan
       - name: Run unit tests
+        if: steps.skiptest.outputs.is-bump == 'no'
         id: unit-test
         run: ./gradlew unitTest --scan
       - name: Run integration tests
+        if: steps.skiptest.outputs.is-bump == 'no'
         id: integration-test
         run: ./gradlew integrationTest --scan
       - name: Upload Test Reports
-        if: always()
+        if: always() && steps.skiptest.outputs.is-bump == 'no'
```

Comment:
this hurts my head. Do you still need the `always()`?

Reply:
Yeah - from some poking around it looks like unless you call `always()` or one of a few similar functions (`success()`, `cancelled()`), [there's an implicit check of `success()`](https://docs.github.com/en/actions/learn-github-actions/expressions#status-check-functions). Calling one of those functions returns a value, but also has the side-effect of overriding this default.

They also hint at that in the [`always()` documentation](https://docs.github.com/en/actions/learn-github-actions/expressions#always) but it would be nice if they explicitly called out that `if: always() && X` behaves differently than `if: X`